desktop/main-app/src/window-manager/abstract.ts

```diff
@@ -46,19 +59,55 @@ export abstract class AbstractWindow {
             ...windowOptions,
         };
 
-        this.win = {
+        const win = {
             options,
             window,
             didFinishLoad: options.isPortal ? Promise.resolve() : window.loadURL(options.url),
         };
 
-        windowOpenDevTools(this.win);
+        if (this.isMultiInstance) {
+            this.wins.push(win);
+        } else {
+            this.wins[0] = win;
+        }
+
+        windowOpenDevTools(win);
 
-        windowHookClose(this.win);
+        windowHookClose(win);
 
-        windowReadyToShow(this.win);
+        windowReadyToShow(win);
 
-        return this.win;
+        return win;
+    }
+
+    public getWin(...ids: MULTI_INSTANCE extends true ? [number] : never[]): CustomWindow | null;
+    public getWin(...ids: any[]): CustomWindow | null {
+        if (this.isEmpty()) {
+            return null;
+        }
+
+        if (this.isMultiInstance) {
+            const id = ids[0];
+            for (const win of this.wins) {
+                if (win.window.id === id) {
+                    return win;
+                }
+            }
+            return null;
+        }
+
+        return this.wins[0] || null;
```

Comment:
```suggestion
        return this.wins?.[0] ?? null;
```

Reply:
It's unnecessary to use `?.` here. The `this.wins` always be an array. We have already declared its type in the class.

```suggestion
        return this.wins[0] ?? null;
```

Reply:
Right, and in the context of the current code, there is no need to use `??`. 
`||` is enough